src/components/search/components/stack.tsx

```diff
@@ -6,63 +6,62 @@ import {loadCourse} from 'lib/courses'
 import {loadLesson} from 'lib/lessons'
 import {loadPlaylist} from 'lib/playlists'
 
+type StackResource = {slug: string; type: string}
+
 type StackProps = {
-  resources: any[]
+  resources: StackResource[]
 }
 
-const Stack: FunctionComponent<StackProps> = ({resources}) => {
-  let res: any[] = []
-  resources.forEach((resource) => {
-    if (resource.type === 'lesson') {
-      const {data: lesson} = useSWR(resource.slug, loadLesson)
-      res.push(lesson)
-    }
-    if (resource.type === 'course') {
-      const {data: course} = useSWR(resource.slug, loadCourse)
-      res.push(course)
-    }
-    if (resource.type === 'playlist') {
-      const {data: playlist} = useSWR(resource.slug, loadPlaylist)
-      res.push(playlist)
-    }
-    return null
-  })
+const StackResource = ({slug, type}: StackResource) => {
+  let loader
+  switch (type) {
+    case 'lesson':
+      loader = loadLesson
+      break
+    case 'course':
+      loader = loadCourse
+      break
+    case 'playlist':
+      loader = loadPlaylist
+      break
+    default:
+      break
+  }
+  const {data: resource} = useSWR(slug, loader)
```

Comment:
I'm not stoked on how this functions on the page because of the loading causing visible shifts. I'd prefer that it was fully static not only for visible aesthetics, but the overall SEO of the page. Since we are opting for convenience here, the good SEO friendly text will be ignored since it is async.

Reply:
Static version of the same thing is here: https://github.com/eggheadio/egghead-next/blob/main/src/components/search/curated/react.tsx#L100

Reply:
So can this be deleted entirely then?